pkg/registry/networking/networkpolicy/strategy.go

```diff
@@ -86,3 +91,25 @@ func (networkPolicyStrategy) ValidateUpdate(ctx context.Context, obj, old runtim
 func (networkPolicyStrategy) AllowUnconditionalUpdate() bool {
 	return true
 }
+
+// Drops Network Policy Disabled fields if Feature Gate is also disabled.
+// This should be used in future Network Policy evolutions
+func dropNetworkPolicyDisabledFields(newNetPol *networking.NetworkPolicy, oldNetPol *networking.NetworkPolicy) {
+	// Clear NetworkPolicyPort EndPort if NetworkPolicyEndPort is not enabled
+	for idx, ingressSpec := range oldNetPol.Spec.Ingress {
```

Comment:
We can optimizie this a bit by returning early when the feature gate is on
```go
endPortGate := utilfeature.DefaultFeatureGate.Enabled(features.NetworkPolicyEndPort)
if endPortGate {
    return
}
...

```

Reply:
Ok, I've changed here. My thinking here was "when we have other new network policy fields that will need also to drop fields, like service selector" instead of early returning, but I think right now we can make this early return and when we get some more new fields, move this logics to inside the ifs